test/jobs/generate_iteration_snippet_job_test.rb

```diff
@@ -49,7 +49,7 @@ class GenerateIterationSnippetJobTest < ActiveJob::TestCase
     create :iteration, solution: @submission.solution
     iteration = create :iteration, submission: @submission
     create :iteration, solution: @submission.solution
-    @submission.solution.update(published_iteration: iteration)
+    @submission.solution.update(published_iteration: iteration, published_at: Time.current)
```

Comment:
The extra `published_at` field needed to be set as that actually determines whether a solution is published